### PR TITLE
Type provider runtime and OAuth manager seams (Fixes #2211)

### LINKED ISSUE #2211: Type provider runtime and OAuth manager seams (Follow-up to #2159)
Follow-up to issue #2159 strict unknown-type audit.

Problem

Provider runtime/composition/OAuth code contains project-owned module seams bridged with unknown casts. Under the stricter audit rule, these are ABUSE because the project controls both sides of the interface and can expose typed methods instead of reaching through private or missing shapes.

Representative audited sites

| path:line | issue |
|---|---|
| packages/providers/src/composition/providerManagerInstance.ts:158 | RuntimeProviderManager is cast to IProviderManager for ProviderContentGenerator. |
| packages/providers/src/composition/providerManagerInstance.ts:250 | OAuthManager private providers map is accessed by casting to an ad-hoc shape. |
| packages/providers/src/composition/providerManagerInstance.ts:512 | ProviderManager constructor is cast to an optional-context constructor shape. |
| packages/providers/src/auth/oauth-manager.ts:81 | OAuthManager self is cast to expose authFlowOrchestrator. |
| packages/providers/src/runtime/runtimeContextFactory.ts:92 | Config is cast to a mutable Pick shape to wire runtime context. |
| packages/providers/src/runtime/runtimeContextFactory.ts:373 | ProviderManager is cast to discover a runtime property. |
| packages/providers/src/runtimeNormalizer.ts:291,294 | Config is cast to discover getSettingsService. |
| packages/providers/src/runtime/profile-application/profileAccessors.ts:14 | Profile is cast to PersistedProfileView instead of parsed/converted. |
| packages/providers/src/auth/proxy/credential-store-factory.ts:83 | Proxy storage is cast to ProviderKeyStorage. |
| packages/providers/src/openai/getOpenAIProviderInfo.ts:139 | Active provider is cast to OpenAIProviderLike. |

Expected fix direction

- Add typed public accessors on OAuthManager and ProviderManager for the needed runtime capabilities.
- Align RuntimeProviderManager/IProviderManager/ProviderContentGenerator contracts so no cast bridge is required.
- Introduce typed provider-runtime context construction APIs instead of mutating config through Pick casts.
- Convert persisted/profile views with a named parser or mapper instead of double-casting.
- Remove unknown casts and do not add lint suppressions.

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

Fixes #2211 — follow-up to the #2159 strict unknown-type audit.

Provider runtime / composition / OAuth / profile / storage code contained **project-owned module seams** bridged with `as unknown as` / `as never` casts. Under the stricter audit rule these are abuse: the project controls both sides of the interface, so it can expose typed methods instead of reaching through private or missing shapes. This PR replaces every enumerated cast bridge with an honest typed contract.

## Audited sites remediated

| path | previous cast | fix |
|---|---|---|
| `composition/providerManagerInstance.ts` | `RuntimeProviderManager` → `IProviderManager` for `ProviderContentGenerator`; OAuthManager private `providers` map cast; `ProviderManager` ctor cast | narrow `ProviderContentGeneratorManager` interface; typed `attachAddItemToProviders`; typed ctor union |
| `auth/oauth-manager.ts` | self-cast to expose `authFlowOrchestrator` | constructor reordered; real typed field + public `attachAddItemToProviders` |
| `runtime/runtimeContextFactory.ts` | `Config` Pick-mutation cast; `ProviderManager` runtime-property cast | typed Config factory setters; `providerManager.setRuntimeContext(scopedRuntime)` |
| `runtimeNormalizer.ts` | `Config` cast to discover `getSettingsService` | direct `config.getSettingsService()` |
| `runtime/profile-application/profileAccessors.ts` | `Profile` double-cast to `PersistedProfileView` | named parser / mapper |
| `auth/proxy/credential-store-factory.ts` | proxy storage cast to `ProviderKeyStorage` | shared `ProviderKeyStorageLike` contract (storage + providers) |
| `openai/getOpenAIProviderInfo.ts` | active provider cast to `OpenAIProviderLike` | exported `OpenAIProviderInfoSource` / `OpenAIProviderLike` contracts |

Supporting changes:

- `core/src/runtime/contracts/RuntimeProviderManager.ts` — contract gains `setRuntimeContext`.
- `core/src/config/configBaseCore.ts` — typed content-generator/tokenizer factory setters + getters (replaces Pick-cast mutation).
- `providers/src/ProviderManager.ts` — implements `setRuntimeContext`; typed optional-context ctor union.
- `providers/src/auth/types.ts` — `OAuthProvider` gains optional `setAddItem`.
- `cli/src/ui/hooks/useOpenAIProviderInfo.ts` — drops `as never`.
- `providers/src/runtime/profileApplication.ts` + `loadBalancerProfile.ts` — downstream `?? ''` fallbacks simplified now that the named accessors return `string`.
- `storage/src/secure-store/provider-key-storage.ts` + `storage/src/index.ts` — export shared `ProviderKeyStorageLike`.

## Test changes

All updated test doubles satisfy the new typed contracts (no escape casts in tests either):

- `core` runtime contract tests + `agents` test-util implement `setRuntimeContext`.
- `oauth-manager.spec.ts` — callbacks typed as `OAuthUICallback`; added an edge-case test proving `attachAddItemToProviders` safely skips providers without `setAddItem`.
- `getOpenAIProviderInfo.context.test.ts` — uses the sanctioned `createRuntimeConfigStub` and real `ConversationCache` instances; typed `OpenAIProviderLike` stubs.
- `runtimeContextFactory.setRuntimeContext.test.ts` (new) — covers the adopted-manager path and stronger assertions.
- `factory-detection-wiring.test.ts` — direct + proxy storage paths both assert the full `ProviderKeyStorageLike` surface.

## Constraints honored

- **Zero** new `as unknown as` / `as any` / `as never` at project-owned sites (including tests) — verified by diff scan and `npm run lint:eslint-guard`.
- **Zero** lint suppressions / `@ts-*` directives; no complexity/size threshold changes; no severity downgrades.
- Behavioral tests only; real instances (no mock theater).

## Verification

- `npm run typecheck` passed
- `npm run lint` passed
- `npm run lint:eslint-guard` passed (ESLint policy guard passed)
- `npm run format` passed (no changes)
- `npm run build` passed
- providers: **3457 passed** / 18 skipped (380 files)
- storage: **311 passed** / 3 skipped (17 files)
- smoke test (`ollamakimi` haiku) passed

### CODE REVIEW

File: packages/providers/src/auth/oauth-manager.ts [hunk 1]
```diff
@@ -114,6 +109,9 @@ export class OAuthManager implements BucketFailoverOAuthManagerLike {
       runtimeDeps?.config,
       runtimeDeps?.messageBus,
     );
+    // Assign the message bus AFTER authFlowOrchestrator exists so the setter's
+    // propagation to the orchestrator takes effect during construction.
+    this._runtimeMessageBus = runtimeDeps?.messageBus;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/auth/oauth-manager.ts]: Fixed in 34033a1b1. You're right — assigning the backing field directly bypassed the setter, so propagation to authFlowOrchestrator never ran during construction (it worked only because the bus is also passed to the orchestrator's constructor). Now routed through the runtimeMessageBus setter; the orchestrator already exists at that point, the call is idempotent with the constructor argument, and propagation stays correct if that argument is ever dropped. The oauth-manager.wiring.spec.ts setter-propagation test and full typecheck/lint/build pass.
